### DISCUSSION TITLE
Where files are temporarily saved when connecting over internet ?

### DISCUSSION THREAD

**Discussion by user858753257** [schlagmichdoch/PairDrop#183] [Q&A]
See tittle ...

**Comment by schlagmichdoch** [comment]
The connection is still the same as before. The only change is, that signaling is now done for devices that areon the same public room and not only between devices with the same public IP address. 

There is no intermediary. Files are never saved anywhere (see index.js). The transferred chunks are saved on the receiving side in memory. Connections are p2p if devices are not behind different NATs. Otherwise, traffic is relayed via the TURN server that is hosted on the same server as the application. Traffic is always encrypted by WebRTC directly. See the technical documentation: https://github.com/schlagmichdoch/PairDrop/blob/master/docs/technical-documentation.md
